app/assets/javascripts/dashboard.js

```diff
@@ -125,7 +130,7 @@ MinionPoller = {
         var hasAdminNodeUpdate = data.admin.update_status === 1 || data.admin.update_status === 2;
         $("#update-all-nodes").toggleClass('hidden', !updateAvailable || hasAdminNodeUpdate);
 
-        MinionPoller.enable_kubeconfig(minions.length > 0 && allApplied);
+        MinionPoller.enable_kubeconfig(minions.length > 0 && masterApplied);
```

Comment:
If `allApplied` and `masterApplied` started with the safest default value (`false`), and then we only set them to `true` inside their special cases if they are met, then this line would just be `MinionPoller.enable_kubeconfig(masterApplied);`